tests/plugins/autocomplete/autocomplete.js

```diff
@@ -219,6 +223,47 @@
 
 			assert.areEqual( '<p>item1</p>', editor.getData() );
 
+			ac.destroy();
+		},
+
+		// (#1987)
+		'test custom view template': function() {
+			var editor = this.editors.standard,
+				itemTemplate = '<li data-id="{id}"><strong>{name}</strong></li>',
+				ac = new CKEDITOR.plugins.autocomplete( editor, matchTestCallback,
+					function( query, range, callback ) {
+						callback( [ { id: 1, name: 'anna' } ] );
+					},
+					itemTemplate );
+
+			this.editorBots.standard.setHtmlWithSelection( '' );
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			assert.areEqual( '<ul><li class="cke_autocomplete_selected" data-id="1"><strong>anna</strong></li></ul>', bender.tools.compatHtml( ac.view.element.getHtml(), false, true ) );
```

Comment:
Instead of nesting different functions, you should use `assert.beautified.html()` here.